tests/unit/test_routes/test_resource_crud.py

```diff
@@ -0,0 +1,552 @@
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.post('/api/v1/resources',
+                           json=1,
+                           headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("int" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.post('/api/v1/resources',
+                           json=1.2,
+                           headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("number" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.post('/api/v1/resources',
+                           json=True,
+                           headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("boolean" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.post('/api/v1/resources',
+                           json="This should be a list",
+                           headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("string" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+
+def test_update_resource_wrong_type(
+        module_client, module_db, fake_auth_from_oc, fake_algolia_save):
+    client = module_client
+    apikey = get_api_key(client)
+    response = client.put('/api/v1/resources/1',
+                          json=[{"expected_error": "This should be a dict"}],
+                          headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("array" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.put('/api/v1/resources/1',
+                          json=1,
+                          headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("int" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.put('/api/v1/resources/1',
+                          json=4.2,
+                          headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("number" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.put('/api/v1/resources/1',
+                          json=True,
+                          headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("boolean" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
+
+    response = client.put('/api/v1/resources/1',
+                          json="This should be a dict",
+                          headers={'x-apikey': apikey})
+    assert (response.status_code == 422)
+    assert ("string" in response.get_json()
+            .get("errors").get("invalid-type").get("message"))
```

Comment:
We could probably DRY up these `wrong_type` tests with a similar utility function. Pass in the expected type and validate that there's a 422 response and that the expected type is in the response message